virtualizarr/readers/hdf/hdf.py

```diff
@@ -218,6 +224,7 @@ def open_virtual_dataset(
             filepath=filepath,
             drop_variables=_drop_vars,
             group=group,
+            store_config=reader_options,
```

Comment:
I just used `reader_options` again here. But we might want to use a different API, see #553